src/services/flow_controller_service.py

```diff
@@ -0,0 +1,38 @@
+from src.schemas.files import PDFFile
+import fitz
+import pytesseract
+from PIL import Image
+from src.exceptions.custom_exceptions import PDFProcessingError
+from src.schemas.files import PDFStatus
+
+
+class FlowService:
+    def __init__(self, pdf_contents: bytes):
+        self.pdf_contents=pdf_contents
+
+    def is_scanned_pdf(self)->bool:
+        try:
+            doc = fitz.open(stream=self.pdf_contents,filetype="pdf")
+            total_pages = doc.page_count
+            num_blank_pages = 0
+            for page_num in range(total_pages):
+                page = doc.load_page(page_num)
+                text = page.get_text()
+                if not text.strip():
+                    num_blank_pages += 1
+            doc.close()
+            if num_blank_pages / total_pages > 0.5:
```

Comment:
if blank pages are scanned then a pdf is searchable if it contains no blank pages, if it contains all blank pages it is not searchable, in the case where it is mixed raise an error with a message: cannot handle mixed pdf